stores/wallet.ts

```diff
@@ -88,14 +177,335 @@ export const useWalletStore = defineStore('wallet', () => {
+            }]
+          : []
+      }
+    }
+  }
+
+  function getEmailAlreadyUsedErrorMessage ({
+    email,
+    evmWallet,
+    likeWallet
+  }: {
+    email: string
+    evmWallet?: string
+    likeWallet?: string
+  }) {
+    if (evmWallet) {
+      return $t('account_register_error_email_already_used_with_evm_wallet', { email, evmWallet })
+    }
+    if (likeWallet) {
+      return $t('account_register_error_email_already_used_with_like_wallet', { email, likeWallet })
+    }
+    return $t('account_register_error_email_already_used', { email })
+  }
+
+  async function register ({
+    walletAddress,
+    email: prefilledEmail,
+    loginMethod,
+    magicUserId,
+    magicDIDToken
+  }: {
+    walletAddress: string
+    email?: string
+    loginMethod: string
+    magicUserId?: string
+    magicDIDToken?: string
+  }) {
+    let tempAccountId = generateAccountIdFromWalletAddress(walletAddress)
+    try {
+      await fetchUserRegisterCheck({ accountId: tempAccountId })
+    } catch (error) {
+      if (error instanceof FetchError && error.data?.error === 'USER_ALREADY_EXIST') {
+        tempAccountId = error.data.alternative as string
+      } else {
+        throw error
+      }
+    }
+
+    const startTime = Date.now()
+    let hasError = false
+    const payload = {
+      accountId: tempAccountId,
+      email: prefilledEmail
+    }
+
+    // Loop until registration is successful, user cancels or timeout
+    do {
+      // Check if registration time exceeds the limit
+      if (Date.now() - startTime > REGISTER_TIME_LIMIT_IN_TS) {
+        throw createError({
+          statusCode: 408,
+          data: { description: $t('account_register_timeout') }
+        })
+      }
+
+      try {
+        // Skip registration modal if email is provided
+        if (!payload.email || hasError) {
+          type ModalResult = { accountId: string; email: string; displayName?: string }
+          let modalResult: ModalResult | null = null
+          // Close login panel first to avoid focus trap
+          bookstoreApiStore.closeLoginPanel()
+          await nextTick()
+          await new Promise<void>((resolve) => {
+            modal.open(RegistrationModal, {
+              email: payload?.email,
+              accountId: payload?.accountId,
+              isAccountIdHidden: true,
+              isDisplayNameHidden: true,
+              onSubmit: (data: ModalResult) => {
+                modalResult = data
+                modal.close()
+                resolve()
+              },
+              onClose: () => {
+                modal.close()
+                resolve()
+              }
+            })
+          })
+          if (!modalResult) {
+            // User canceled the registration, return false immediately
+            return false
+          }
+          const result = modalResult as ModalResult
+          payload.accountId = result.accountId
+          payload.email = result.email
+        }
+        hasError = false
```

Comment:
`hasError` is overloaded here, it is both used as condition for showing `RegistrationModal` with `email`, and condition to exit a huge loop.
Would be better if we can use 2 diff variable (even if they would have same values) to make the intention clear